pkg/registry/minion/etcd/etcd.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	etcdgeneric "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/minion"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/tools"
+)
+
+type REST struct {
+	*etcdgeneric.Etcd
+}
+
+// NewREST returns a RESTStorage object that will work against nodes.
+func NewREST(h tools.EtcdHelper) *REST {
+	prefix := "/registry/minions"
+	store := &etcdgeneric.Etcd{
+		NewFunc:     func() runtime.Object { return &api.Node{} },
+		NewListFunc: func() runtime.Object { return &api.NodeList{} },
+		KeyRootFunc: func(ctx api.Context) string {
+			return prefix
+		},
+		KeyFunc: func(ctx api.Context, name string) (string, error) {
+			return prefix + "/" + name, nil
+		},
+		ObjectNameFunc: func(obj runtime.Object) (string, error) {
+			return obj.(*api.Node).Name, nil
+		},
+		PredicateFunc: func(label labels.Selector, field fields.Selector) generic.Matcher {
+			return minion.MatchNode(label, field)
+		},
+		EndpointName: "minion",
+
+		Helper: h,
+	}
+
+	store.CreateStrategy = minion.Strategy
```

Comment:
Took me a while to digest why we need both a rest and registry looking at the code today because if we plug in a RESTStorage against a url, nothing really calls into the registry. I guess it:

1. Makes the client and server interfaces similar, because they're both returning non-generic types
2. Allows *other* services to access the resource using a slightly higher level of abstraction (eg: rc for listing pods, podcache).

Correct? fyi I'm porting rcs to genericetcd.